platform/darwin/src/MGLMultiPoint.mm

```diff
@@ -27,6 +58,61 @@ - (instancetype)initWithCoordinates:(const CLLocationCoordinate2D *)coords count
     return self;
 }
 
+- (instancetype)initWithCoder:(NSCoder *)decoder
+{
+    self = [super initWithCoder:decoder];
+    if (self)
+    {
+        NSArray *coordinates = [decoder decodeObjectOfClass:[NSArray class] forKey:@"coordinates"];
+        CLLocationCoordinate2D *coords = (CLLocationCoordinate2D *)malloc([coordinates count] * sizeof(CLLocationCoordinate2D));
+        for (NSUInteger i = 0; i < [coordinates count]; i++)
+        {
+            coords[i] = CLLocationCoordinate2DMake([coordinates[i][0] doubleValue], [coordinates[i][1] doubleValue]);
+        }
+        _coordinates = { coords, coords + [coordinates count] };
+        free(coords);
+    }
+    return self;
+}
+
+- (void)encodeWithCoder:(NSCoder *)coder
+{
+    [super encodeWithCoder:coder];
+    
+    NSMutableArray *coordinates = [NSMutableArray array];
+    for (auto coord : _coordinates) {
+        [coordinates addObject:@[@(coord.latitude), @(coord.longitude)]];
+    }
+    [coder encodeObject:coordinates forKey:@"coordinates"];
+}
+
+- (BOOL)isEqual:(id)other
+{
+    if (self == other) return YES;
+    
+    MGLMultiPoint *otherMultipoint = other;
+    
+    CLLocationCoordinate2D *otherCoordinates = [otherMultipoint coordinates];
+    std::vector<CLLocationCoordinate2D> otherCoords;
+    otherCoords.reserve([otherMultipoint pointCount]);
+    
+    for (NSUInteger i = 0; i < [otherMultipoint pointCount]; i++) {
+        otherCoords.push_back(otherCoordinates[i]);
+    }
+    
+    return ([super isEqual:otherMultipoint]
+            && _coordinates == otherCoords);
```

Comment:
This could be written as `_coordinates == otherCoords->_coordinates` to avoid having to build a copy of the vector in the first place.